instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/OpenTelemetryAutoConfiguration.java

```diff
@@ -57,18 +57,20 @@ public class OpenTelemetryAutoConfiguration {
 
   public OpenTelemetryAutoConfiguration() {}
 
+  @Bean
+  @ConfigurationPropertiesBinding
+  public MapConverter mapConverter() {
+    // needed for otlp exporter headers and OtelResourceProperties
```

Comment:
Have you considered to add a no-op map converter bean  in `DisabledOpenTelemetrySdkConfig`?

```java
public class NoOpMapConverter implements Converter<String, Map<String, String>> {

  @Override
  public Map<String, String> convert(String source) {
    return Collections.emptyMap();
  }
}
```

Reply:
what would be the benefit?

Reply:
it's actually dangerous - the user might accidentally rely on the map converter - so we should not change it when disabling the SDK

Reply:
The benefit is to avoid executing OTel code if the SDK is disabled. Imagine that you have a bug in `MapConverter` that prevents the application from starting.

Reply:
> it's actually dangerous - the user might accidentally rely on the map converter - so we should not change it when disabling the SDK

I don't follow. The `MapConverter` bean is no part of the public API.

Reply:
that's right - but it's a behavior that can also affect other parts of the spring application.

In other words, a user might use `a=b,c=d` for their own map properties.

Reply:
The OTel map converter could only parse OTel properties.

In addition, `mapConverter` is a quite generic bean name. I would suggest something more specific to OTel, for example `otelMapConverter`.

Reply:
> The OTel map converter could only parse OTel properties.

that's not how it works - it's registered globally in spring

Reply:
Having said that, what do you think about:
*  Fix the bug with this PR code without the API change added in a second commit
*  Create an issue for potential additional changes